crates/l2/contracts/src/l1/OnChainProposer.sol

```diff
@@ -1,14 +1,68 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.27;
 
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
 import "./interfaces/IOnChainProposer.sol";
+import {CommonBridge} from "./CommonBridge.sol";
+import {ICommonBridge} from "./interfaces/ICommonBridge.sol";
 
 /// @title OnChainProposer contract.
 /// @author LambdaClass
-contract OnChainProposer is IOnChainProposer {
+contract OnChainProposer is IOnChainProposer, ReentrancyGuard {
+    /// @notice The commitments of the committed blocks.
+    /// @dev If a block is committed, the commitment is stored here.
+    /// @dev If a block was not committed yet, it won't be here.
+    /// @dev It is used by other contracts to verify if a block was committed.
+    mapping(uint256 => bytes32) public blockCommitments;
+
+    /// @notice The verified blocks.
+    /// @dev If a block is verified, the block hash is stored here.
+    /// @dev If a block was not verified yet, it won't be here.
+    /// @dev It is used by other contracts to verify if a block was verified.
+    mapping(uint256 => bool) public verifiedBlocks;
+
+    address public BRIDGE;
+
+    constructor() {}
+
+    function initialize(address bridge) public nonReentrant {
```

Comment:
The same here about only call this method once

Reply:
Done [285f41c](https://github.com/lambdaclass/lambda_ethereum_rust/pull/944/commits/285f41c8326b1899788ad0b8313477741f5313ca)